src/base_txn.cpp

```diff
@@ -218,9 +218,9 @@ namespace nogdb {
                         WriteLock<boost::shared_mutex> _(*(ctx.dbInfoMutex));
                         (*ctx.dbInfo) = dbInfo;
                     }
-                    // allow the next txns to see the latest version and updates
-                    ctx.dbTxnStat->fetchAddMaxVersionId();
                 }
+                // allow the next txns to see the latest version and updates
+                ctx.dbTxnStat->fetchAddMaxVersionId();
```

Comment:
Why need to move this?

Reply:
So, `txn version` will get increased even though the graph has not been changed. `txn version` plays an important role in record version updates.

Reply:
Good point. don't forget to fix the failure test according to this change.